### PR TITLE
fix(user): force locked achievement tooltips on user pages to grayscale

### PR DESCRIPTION
**Pull request by wescopeland**
This PR addresses an issue raised in https://github.com/RetroAchievements/RAWeb/pull/1487 where locked achievement badges shown in tooltips on user profile pages are not in grayscale.

I have come to believe this behavior is a bug, mostly due to badge artwork like what was created for [Myst on 3DO](http://retroachievements.org/game/16839).

This PR restores these images to be grayscale until the achievements are unlocked by the user.

https://user-images.githubusercontent.com/3984985/233530759-9cbc2a91-4ccc-4e4f-99a9-1ce4cd431aac.mp4

### CODE REVIEW

**Comment by Jamiras** [comment]
This has the side effect of making all embedded achievement tooltips disabled, regardless of whether or not the player has unlocked them.

image

I think it's desirable to show the colorized icon in the embed itself. For now, the tooltip for an embedded achievement should also be colorized. Maybe in the future, we can include the user unlock state there as well.

This is most likely why the tooltips on the profile page always show colored.

**Comment by wescopeland** [comment]
> I think it's desirable to show the colorized icon in the embed itself. For now, the tooltip for an embedded achievement should also be colorized. Maybe in the future, we can include the user unlock state there as well.

@Jamiras I agree. I made a mistake in my original implementation.

I have now updated this implementation such that only the embeds on the user profile are greyscale.

File: app_legacy/Helpers/render/achievement.php [hunk 1]
```diff
@@ -101,10 +103,11 @@ function renderAchievementCard(int|string|array $achievement, ?string $context =
     $achPoints = $data['Points'] ?? null;
     $badgeName = $data['BadgeName'] ?? null;
     $unlock = $data['Unlock'] ?? null;
+    $badgeImgSrc = media_asset("Badge/{$badgeName}" . ($shouldForceLockedImage ? '_lock' : '') . ".png");
```
Comment (Jamiras) [inline app_legacy/Helpers/render/achievement.php:106]: Wouldn't it be easier to just pass `$icon` through and change this to:
```
$badgeImageSrc = $icon ?? media_asset("Badge/{$badgeName}.png");
```

Comment (wescopeland) [reply app_legacy/Helpers/render/achievement.php]: Quite a bit easier 😊

I've made this change in https://github.com/RetroAchievements/RAWeb/pull/1498/commits/425d56b5f0fb895b3146c6812934964ce997f226.

I have kept the variable name as `$badgeImgSrc` to slightly imply it's being attached to the `src` attribute of an `` tag.

The argument name I've set as `$iconUrl`.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**